src/components/PeoplePage.tsx

```diff
@@ -1,35 +1,152 @@
+  const filterBySex = searchParams.get('sex') || null;
+  const centuriesOption = searchParams.getAll('centuries') || [];
+  const sortByUrl = searchParams.get('sort') || null;
+  const sortByOrder = searchParams.get('order') || null;
+  const { personSlug } = useParams();
+
+  let filteredPeople = [...people];
+
+  useEffect(() => {
+    const fetchData = async () => {
+      try {
+        const getPeopleFromServer = await getPeople();
+
+        const peopleFromServer = getPeopleFromServer.map(person => {
+          const mother = getPeopleFromServer.find(p => p.name === person.motherName);
+          const father = getPeopleFromServer.find(p => p.name === person.fatherName);
+
+          return {
+            ...person,
+            mother,
+            father,
+          };
+        });
+
+        setPeople(peopleFromServer);
+      } catch {
+        setIsError('Something went wrong');
+      } finally {
+        setIsLoading(false);
+      }
+    };
+
+    fetchData();
+  }, []);
+
+  if (filterBySex) {
+    filteredPeople = filteredPeople.filter(person => person.sex === filterBySex);
+  }
+
+  useEffect(() => {
+    if (!query) {
+      const updatedSearchParams = getSearchWith(searchParams, { query: null });
 
-          <div className="column">
-            <div className="box table-container">
-              <Loader />
+      setSearchParams(updatedSearchParams);
+    }
+  }, [query, searchParams, setSearchParams]);
 
-              <p data-cy="peopleLoadingError">Something went wrong</p>
+  if (query) {
+    filteredPeople = filteredPeople.filter(person => {
+      return person.name.toLowerCase().includes(query.toLowerCase())
+        || (person.motherName && person.motherName.toLowerCase().includes(query.toLowerCase()))
+        || (person.fatherName && person.fatherName.toLowerCase().includes(query.toLowerCase()));
+    });
+  }
 
-              <p data-cy="noPeopleMessage">
-                There are no people on the server
-              </p>
+  if (centuriesOption.length > 0) {
+    filteredPeople = filteredPeople.filter((person) => {
+      const alivePeriod = Math.ceil(person.born / 100);
+
+      return centuriesOption.includes(alivePeriod.toString());
+    });
+  }
+
+  if (sortByUrl) {
+    filteredPeople = filteredPeople.sort((a, b) => {
+      switch (sortByUrl) {
+        case 'name': return a.name.localeCompare(b.name);
+
+        case 'sex': return a.sex.localeCompare(b.sex);
+
+        case 'born': return a.born - b.born;
+
+        default: return a.died - b.died;
+      }
+    });
+
+    if (sortByOrder) {
+      filteredPeople.reverse();
+    }
+  }
+
+  return (
+    <div className="section">
+      <div className="container">
+        <h1 className="title">People Page</h1>
+
+        <div className="block">
+          <div className="columns is-desktop is-flex-direction-row-reverse">
+            <div className="column is-7-tablet is-narrow-desktop">
+              <PeopleFilters
```

Comment:
Add conditional render to this component depending on the people's data. 

- The sidebar with the filters should appear only when people are loaded.